fvm/meter/weighted/meter.go

```diff
@@ -153,16 +283,21 @@ func (m *Meter) ComputationIntensities() interfaceMeter.MeteredComputationIntens
 
 // TotalComputationUsed returns the total computation used
 func (m *Meter) TotalComputationUsed() uint {
-	return uint(m.computationUsed >> MeterInternalPrecisionBytes)
+	return uint(m.computationUsed >> MeterExecutionInternalPrecisionBytes)
 }
 
 // TotalComputationLimit returns the total computation limit
 func (m *Meter) TotalComputationLimit() uint {
-	return uint(m.computationLimit >> MeterInternalPrecisionBytes)
+	return uint(m.computationLimit >> MeterExecutionInternalPrecisionBytes)
 }
 
 // SetMemoryWeights sets the memory weights
 func (m *Meter) SetMemoryWeights(weights ExecutionMemoryWeights) {
+	if int(common.MemoryKindLast)-len(weights) != 1 {
+		for kind, weight := range weights {
+			m.memoryWeights[kind] = weight
+		}
+	}
 	m.memoryWeights = weights
```

Comment:
This is doing a bit of work then throwing it out with `m.memoryWeights = weights`. It also might be overwriting `DefaultMemoryWeights`.

Reply:
Right yeah, this is meant to be an else